components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/pap/PAPPolicyReader.java

```diff
@@ -62,14 +69,22 @@
 
     private PAPPolicyReader(PolicyFinder policyFinder) {
 
+
         this.policyFinder = policyFinder;
         // create the factory
-        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
-        factory.setIgnoringComments(true);
-        factory.setNamespaceAware(true);
+        DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
+        documentBuilderFactory.setIgnoringComments(true);
+        documentBuilderFactory.setNamespaceAware(true);
+        documentBuilderFactory.setExpandEntityReferences(false);
+        SecurityManager securityManager = new SecurityManager();
+        securityManager.setEntityExpansionLimit(ENTITY_EXPANSION_LIMIT);
+        documentBuilderFactory.setAttribute(SECURITY_MANAGER_PROPERTY, securityManager);
+
         // now use the factory to create the document builder
         try {
-            builder = factory.newDocumentBuilder();
+            documentBuilderFactory.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+            builder = documentBuilderFactory.newDocumentBuilder();
+            builder.setEntityResolver(new CarbonEntityResolver());
             builder.setErrorHandler(this);
         } catch (ParserConfigurationException pce) {
             throw new IllegalArgumentException("Filed to setup repository: ");
```

Comment:
exception object is not included in the wrapped exception. Provide some context information in the message.